components/confirm_modal.jsx

```diff
@@ -93,6 +93,10 @@ export default class ConfirmModal extends React.Component {
         document.removeEventListener('keydown', this.handleKeypress);
     }
 
+    shouldComponentUpdate(nextProps) {
```

Comment:
At a high level, I'm thinking we need to be cautious about using `PureComponent` as a general solution and expecting performance improvements (e.g. see https://twitter.com/dan_abramov/status/820672447418605568). We'd have to profile it to be sure, but the `PureComponent` implementation of `shouldComponentUpdate` checking 10+ props might be on the order of just doing this particular `render()` in the first place. I'm also a little fuzzy on whether the props being passed into this component would even pass a shallow quality test (given some of them are `const` and declared on the stack.)

That aside, however, the intent here is to skip aggressively and only update the component when it should be rendered. This was happening for each character being typed, so even a `PureComponent` would shallow check all these properties each letter of a message. 

I'd actually prefer to write this directly into `CreatePost` (and elsewhere) as:
```
{this.state.showConfirmModal &&
                <ConfirmModal
                    title={notifyAllTitle}
                    message={notifyAllMessage}
                    confirmButtonText={notifyAllConfirm}
                    show={this.state.showConfirmModal}
                    onConfirm={this.handleNotifyAllConfirmation}
                    onCancel={this.hideNotifyAllModal}
                />
}
```

skipping the initial mount altogether and clearly expressing the intent here, without resorting to the mechanics of `shouldComponentUpdate`.

Thoughts?

Reply:
Agreed re: `shouldComponentUpdate`: measure, measure, measure, before over optimizing.

I'd love to understand more your opinion about the parent choosing to render vs. the child. To me, the `{show && <child />}` paradigm is idiomatic, and works with any child component, react-bootstrap or otherwise. That being said, it doesn't work if a child component wants to animate in or out: obviously it needs to be "running" to control that. Maybe chat at dev hangout tomorrow?

Reply:
Dev hangout is today actually 🙃, but yeah, we can definitely discuss this more since I feel like I've mentioned this logic occasionally to different people without ensuring other people actually know what I'm talking about.